proxy/httpp/interface.go

```diff
@@ -120,6 +138,24 @@ func (t *Transform) Apply(req *http.Request) bool {
 		req.Header["X-Forwarded-For"] = nil
 	case XForwardedForAdd:
 	}
+
+	switch t.XWebauth {
+	case "":
+		fallthrough
+	case XWebauthNone:
+		// No extra headers set
+	case XWebauthSet:
+		creds := oauth.GetCredentials(req.Context())
+		if creds == nil {
+			// TODO: any opportunity to log here? Metrics?
```

Comment:
can we get a jira ticket for this - this will hide issues and we really should get some sort of logging here for this - or refactor this method to return (bool,error)

Reply:
yeah I'm hoping @ccontavalli can clue me into if a logger/metrics is accessible here? and I can fix it in this PR

Reply:
Need to extend the Transform object to carry metrics/logs, or (arguably better?) pass it down to Apply? extend contract of Apply so it can eg, fail, or log or increment counters.

That said, if the mapping has authentication required, you'll never get here without credentials being available? And if mapping does NOT have authentication (marked as public), then having WebAuth set to Set should fail config validation, and return an error there? So we can never end up silently in this condition? This PR is still a good step in the right direction.

Reply:
Passed the logger in via context instead - so it gets plumbed from Proxy -> AuthenticatedProxy, which puts it in the context of each request. Fetching the logger from the context returns a NilLogger if none is present - so a fallback to noop/safe behavior.